drivers/serial/uart_ns16550.c

```diff
@@ -251,6 +252,10 @@ BUILD_ASSERT_MSG(IS_ENABLED(CONFIG_PCIE), "NS16550(s) in DT need CONFIG_PCIE");
 #define OUTBYTE(x, d) OUTWORD(x, d)
 #endif
 
+#ifdef CONFIG_UART_INTERRUPT_DRIVEN
+	static struct k_spinlock lock;
```

Comment:
shouldn't this spinlock be per-device instead of global? I would really expect the lock to be a member of uart_ns16550_dev_data_t

Reply:
Yes, thank you! I'll update that.